src/tools/schema2cpp.ts

```diff
@@ -0,0 +1,191 @@
+/**
+ * @license
+ * Copyright (c) 2019 Google Inc. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+import {Schema2Base} from './schema2base.js';
+import {Schema} from '../runtime/schema.js';
+
+// https://en.cppreference.com/w/cpp/keyword
+// [...document.getElementsByClassName('wikitable')[0].getElementsByTagName('code')].map(x => x.innerHTML);
+const keywords = [
+  'alignas', 'alignof', 'and', 'and_eq', 'asm', 'auto', 'bitand', 'bitor', 'bool', 'break', 'case',
+  'catch', 'char', 'char8_t', 'char16_t', 'char32_t', 'class', 'compl', 'concept', 'const',
+  'consteval', 'constexpr', 'const_cast', 'continue', 'co_await', 'co_return', 'co_yield',
+  'decltype', 'default', 'delete', 'do', 'double', 'dynamic_cast', 'else', 'enum', 'explicit',
+  'export', 'extern', 'false', 'float', 'for', 'friend', 'goto', 'if', 'inline', 'int', 'long',
+  'mutable', 'namespace', 'new', 'noexcept', 'not', 'not_eq', 'nullptr', 'operator', 'or', 'or_eq',
+  'private', 'protected', 'public', 'reflexpr', 'register', 'reinterpret_cast', 'requires',
+  'return', 'short', 'signed', 'sizeof', 'static', 'static_assert', 'static_cast', 'struct',
+  'switch', 'template', 'this', 'thread_local', 'throw', 'true', 'try', 'typedef', 'typeid',
+  'typename', 'union', 'unsigned', 'using', 'virtual', 'void', 'volatile', 'wchar_t', 'while',
+  'xor', 'xor_eq'
+];
+
+// type-char to [cpp-type, pass-by-reference]
+const typeMap = {
+  'T': ['std::string', true],
+  'U': ['URL', true],
+  'N': ['double', false],
+  'B': ['bool', false],
+};
+
+export class Schema2Cpp extends Schema2Base {
+  // test-CPP.file_name.manifest -> test-cpp-file-name-manifest.h
+  outputName(baseName: string): string {
```

Comment:
Maybe add a few unit tests here as sanity checks that the output name will always be correct?

Reply:
I want to add a full suite of unit tests for these, but need to prioritise other things first.